nixos/modules/services/networking/keepalived/default.nix

```
@@ -282,16 +295,25 @@ in
       };
     };
 
-    systemd.services.keepalived = {
+    systemd.services.keepalived = let
+      finalConfigFile = if cfg.secretFile == null then keepalivedConf else "/var/run/keepalived/keepalived.conf";
```

```suggestion
      finalConfigFile = if cfg.secretFile == null then keepalivedConf else "/run/keepalived/keepalived.conf";
```

/var/run is just a symlink these days.